### PR TITLE
Refactoring: Removing Unused Default Values and Unnecessary Timeout Field

### PR DESCRIPTION
**Pull request by msadeqsirjani**
## Changes

- You don't need to set an infinite timeout for this method since the default value is already set to -1, which means the timeout is infinite.
- The exporterTimeoutMilliseconds field is no longer used in the BatchExportProcessor class. It has been removed to simplify the code and remove unnecessary complexity.
- The parentheses in the constructor of the Thread class are not necessary and can be removed. This improves the readability of the code and reduces the number of unnecessary characters.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)

### CODE REVIEW

File: src/OpenTelemetry/BatchExportProcessor.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public abstract class BatchExportProcessor<T> : BaseExportProcessor<T>
             this.scheduledDelayMilliseconds = scheduledDelayMilliseconds;
             this.exporterTimeoutMilliseconds = exporterTimeoutMilliseconds;
             this.MaxExportBatchSize = maxExportBatchSize;
-            this.exporterThread = new Thread(new ThreadStart(this.ExporterProc))
+            this.exporterThread = new Thread(this.ExporterProc)
```
Comment (reyang) [inline src/OpenTelemetry/BatchExportProcessor.cs:75]: 👍

**Comment by cijothomas** [comment]
@msadeqsirjani Could you clear the CLA, which is required for any contributions to OpenTelemetry repos. Thank you!

**Comment by msadeqsirjani** [comment]
@cijothomas CLA is configured right now.

**Comment by msadeqsirjani** [comment]
> @msadeqsirjani Could you clear the CLA, which is required for any contributions to OpenTelemetry repos. Thank you!

CLA is configured right now.

**Review by reyang — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by msadeqsirjani** [comment]
@cijothomas @reyang When will this PR be merged with the main branch?

**Comment by reyang** [comment]
> @cijothomas @reyang When will this PR be merged with the main branch?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md#how-to-get-prs-merged

File: src/OpenTelemetry/BatchExportProcessor.cs [hunk 2]
```diff
@@ -38,7 +38,6 @@ public abstract class BatchExportProcessor<T> : BaseExportProcessor<T>
 
         private readonly CircularBuffer<T> circularBuffer;
         private readonly int scheduledDelayMilliseconds;
-        private readonly int exporterTimeoutMilliseconds;
```
Comment (CodeBlanch) [inline src/OpenTelemetry/BatchExportProcessor.cs:41]: @alanwest @cijothomas @utpilla Is this indicative of a bug? Should we be using this value somewhere?

Comment (vishweshbankwar) [reply src/OpenTelemetry/BatchExportProcessor.cs:41]: missing [feature](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor:~:text=exportTimeoutMillis%20%2D%20how%20long%20the%20export%20can%20run%20before%20it%20is%20cancelled.%20The%20default%20value%20is%2030000.)?

Comment (utpilla) [reply src/OpenTelemetry/BatchExportProcessor.cs:41]: > missing [feature](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor:~:text=exportTimeoutMillis%20%2D%20how%20long%20the%20export%20can%20run%20before%20it%20is%20cancelled.%20The%20default%20value%20is%2030000.)?

Looks like it. We do let the user configure this value for the exporters.

**Comment by utpilla** [comment]
@msadeqsirjani Could you please update the PR description to reflect the actual changes being made in the PR?

> * The null check before returning the LogRecord to the pool is unnecessary as the Current property of the enumerator is guaranteed to be non-null. Removing the null check improves code readability and maintainability.

Remove this from the description. This is not a part of this PR. Moreover, this check is probably required.

> * The null check before disposing of is unnecessary as the Dispose method is safe to call on null objects. This change simplifies the code and improves readability.

It's not safe to call Dispose on null objects. This change works because `cur.Value` will not be `null`.

> * The infinite timeout was causing the application to hang indefinitely when shutting down. Removing the timeout parameter from the Shutdown method resolves the issue.

That's not what it's doing. This change just removes the explicitly passed method argument which was anyway set to the default value.

**Comment by msadeqsirjani** [comment]
> @msadeqsirjani Could you please update the PR description to reflect the actual changes being made in the PR?
> 
> > * The null check before returning the LogRecord to the pool is unnecessary as the Current property of the enumerator is guaranteed to be non-null. Removing the null check improves code readability and maintainability.
> 
> Remove this from the description. This is not a part of this PR. Moreover, this check is probably required.
> 
> > * The null check before disposing of is unnecessary as the Dispose method is safe to call on null objects. This change simplifies the code and improves readability.
> 
> It's not safe to call Dispose on null objects. This change works because `cur.Value` will not be `null`.
> 
> > * The infinite timeout was causing the application to hang indefinitely when shutting down. Removing the timeout parameter from the Shutdown method resolves the issue.
> 
> That's not what it's doing. This change just removes the explicitly passed method argument which was anyway set to the default value.

The first two items are essential and I have modified them after conducting tests. However, the third option represents an enhancement to the existing features. Therefore, I propose updating the PR description accordingly to reflect these changes.

File: src/OpenTelemetry/BatchExportProcessor.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry/BatchExportProcessor.cs:41]: https://github.com/open-telemetry/opentelemetry-dotnet/issues/1735

Comment (msadeqsirjani) [reply src/OpenTelemetry/BatchExportProcessor.cs]: I revert this commit
Commit SHA (8380e31c1665e1458e03eb85905f11e44d8b7bac)

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
> The exporterTimeoutMilliseconds field is no longer used in the BatchExportProcessor class. It has been removed to simplify the code and remove unnecessary complexity.

@msadeqsirjani Please update the description to remove this entry.

**Pull request merged**